app/models/labware.rb

```diff
@@ -153,6 +168,15 @@ def ancestors_of_purpose(ancestor_purpose_id)
     ancestors.order(id: :desc).where(plate_purpose_id: ancestor_purpose_id)
   end
 
+  # Gets the relevant SpikedBuffer tube, if one exists, by using the two associations.
+  # A direct parent SpikedBuffer tube is used if it exists, otherwise the most recent ancestor.
+  # This was necessary to avoid affecting historical data, for which the direct parent should be used,
+  # even though there is another ancestor that was created more recently.
+  def spiked_in_buffer
```

Comment:
If you still define an association with the buffer itsef this can become simply
```ruby
direct_spiked_in_buffer || most_recent_spiked_in_buffer
```